spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/JGitEnvironmentRepository.java

```diff
@@ -430,20 +442,26 @@ protected void configure(Host hc, Session session) {
 		}
 	}
 
-	private void setCredentialsProvider(TransportCommand<?, ?> cmd) {
-		if (gitCredentialsProvider != null) {
-			cmd.setCredentialsProvider(gitCredentialsProvider);
-		} else if (hasText(getUsername())) {
-			cmd.setCredentialsProvider(
-					new UsernamePasswordCredentialsProvider(getUsername(), getPassword()));
-		} else if (hasText(getPassphrase())) {
-			cmd.setCredentialsProvider(
-					new PassphraseCredentialsProvider(getPassphrase()));
-		}
+	private void configureCommand(TransportCommand<?, ?> command) {
+		command.setTimeout(this.timeout);
+		command.setTransportConfigCallback(this.transportConfigCallback);
+		command.setCredentialsProvider(getCredentialsProvider());
```

Comment:
This needs to be protected by a null check, since that is what was done previously.